core/translate/schema.rs

```diff
@@ -767,11 +767,183 @@ pub fn translate_drop_table(
     let r7 = program.alloc_register();
     program.emit_null(r6, Some(r7));
 
-    //  3. TODO: Open an ephemeral table, and read over triggers from schema table into ephemeral table
-    //  Requires support via https://github.com/tursodatabase/limbo/pull/768
+    //  3. Open an ephemeral table, and read over entries from the schema table for entries of the table whose root page was moved in the destroy operation
+
+    //  cursor id 1
+    let sqlite_schema_cursor_id_1 = program.alloc_cursor_id(
+        Some(SQLITE_TABLEID.to_owned()),
+        CursorType::BTreeTable(schema_table.clone()),
+    );
+    let simple_table_rc = Rc::new(BTreeTable {
+        root_page: 0, // Not relevant for ephemeral table definition
+        name: "ephemeral_scratch".to_string(),
+        has_rowid: true,
+        primary_key_columns: vec![],
+        columns: vec![Column {
+            name: Some("rowid".to_string()),
+            ty: Type::Integer,
+            ty_str: "INTEGER".to_string(),
+            primary_key: false,
+            is_rowid_alias: false,
+            notnull: false,
+            default: None,
+            unique: false,
+            collation: None,
+        }],
+        is_strict: false,
+        unique_sets: None,
+    });
+    //  cursor id 2
+    let ephemeral_cursor_id = program.alloc_cursor_id(
+        Some("scratch_table".to_string()),
+        CursorType::BTreeTable(simple_table_rc),
+    );
+    program.emit_insn(Insn::OpenEphemeral {
+        cursor_id: ephemeral_cursor_id,
+        is_table: true,
+    });
+    let if_not_label = program.allocate_label();
+    program.emit_insn(Insn::IfNot {
+        reg: r2_reg,
+        target_pc: if_not_label,
+        jump_if_null: true, //  jump anyway
+    });
+    program.emit_insn(Insn::OpenRead {
+        cursor_id: sqlite_schema_cursor_id_1,
+        root_page: 1usize.into(),
+    });
+
+    let r8 = program.alloc_register();
+    let r9 = program.alloc_register();
+    let r10 = program.alloc_register();
+    let r11 = program.alloc_register();
+    let r12 = program.alloc_register();
+    let r13 = program.alloc_register();
+    let _r14 = program.alloc_register(); //  Unsure why this register is allocated but putting it in here to make comparison with SQLite easier
+    let r15 = program.alloc_register();
+
+    //  Loop to copy over row id's from the schema table for rows that have the same root page as the one that was moved
+    let end_metadata_label = program.allocate_label();
+    let metadata_loop = program.allocate_label();
```

Comment:
can we call these `moved_rootpage_copy_loop` or something -- unsure what "metadata" means here